appcues/src/main/java/com/appcues/trait/appcues/TooltipTrait.kt

```diff
@@ -0,0 +1,164 @@
+package com.appcues.trait.appcues
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredSizeIn
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.GenericShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.max
+import com.appcues.data.model.AppcuesConfigMap
+import com.appcues.data.model.getConfigStyle
+import com.appcues.trait.AppcuesTraitAnimatedVisibility
+import com.appcues.trait.ContentWrappingTrait
+import com.appcues.trait.PresentingTrait
+import com.appcues.trait.appcues.TooltipTrait.PointerPosition.BOTTOM
+import com.appcues.trait.appcues.TooltipTrait.PointerPosition.NONE
+import com.appcues.trait.appcues.TooltipTrait.PointerPosition.TOP
+import com.appcues.ui.AppcuesOverlayViewManager
+import com.appcues.ui.composables.LocalAppcuesStepMetadata
+import com.appcues.ui.composables.rememberAppcuesContentVisibility
+import com.appcues.ui.extensions.styleBackground
+import com.appcues.ui.modal.dialogEnterTransition
+import com.appcues.ui.modal.dialogExitTransition
+import com.appcues.ui.utils.AppcuesWindowInfo
+import com.appcues.ui.utils.rememberAppcuesWindowInfo
+import org.koin.core.scope.Scope
+
+internal class TooltipTrait(
+    override val config: AppcuesConfigMap,
+    val scope: Scope,
+) : PresentingTrait, ContentWrappingTrait {
+
+    companion object {
+
+        const val TYPE = "@appcues/tooltip"
+
+        private val SCREEN_PADDING = 16.dp
+        private val MAX_WIDTH_DP = 350.dp
+        private val MAX_HEIGHT_DP = 200.dp
+    }
+
+    private enum class PointerPosition {
+        TOP, BOTTOM, NONE
```

Comment:
none is when a "rect" is not present so we know we should position the tooltip at the bottom (for now) and also that we should not compose the pointer at all